lib/Dialect/AIEX/Transforms/AIEDmaToIpu.cpp

```diff
@@ -335,6 +335,37 @@ struct DmaToIpuPattern : OpConversionPattern<IpuDmaMemcpyNdOp> {
   }
 };
 
+/// Convert IpuDmaWaitOp into IpuSyncOp by retrieving the necessary
+/// information from the ShimDMAAllocationOp referenced through the
+/// symbol argument of this op.
+struct DmaWaitToIpuPattern : OpConversionPattern<IpuDmaWaitOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  DmaWaitToIpuPattern(MLIRContext *context, PatternBenefit benefit = 1)
+      : OpConversionPattern(context, benefit) {}
+
+  LogicalResult
+  matchAndRewrite(IpuDmaWaitOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto dev = op->getParentOfType<AIE::DeviceOp>();
+    auto shimDmaAllocOp = getAllocOpForSymbol(dev, op.getSymbol());
+    if (!shimDmaAllocOp) {
+      op.emitOpError("couldn't find shim_dma_allocation op");
+      return failure();
+    }
+    auto channelDir = shimDmaAllocOp->getChannelDir();
+    int channel = shimDmaAllocOp->getChannelIndex();
+    int direction = (int)(channelDir == AIE::DMAChannelDir::MM2S);
+    int column = shimDmaAllocOp->getCol();
+
+    // Create with `column_num == 1` and `row_num == 1` to check for a single
```

Comment:
I guess this is reasonable (always setting both of these to 1) but I think the signal here is that actually `IpuSyncOp` just shouldn't expose these attributes (and then in the write32s set these to 1).

Reply:
Maybe it's ok in `IpuSyncOp`. I have to look at the details again, but you could use this as a `wait_all` I think? For example increase `col_num` to wait for multiple TCTs from different columns? A `DmaWaitAllOp` could be added to accommodate that if needed.

Reply:
Ideally, `ipu_sync` wouldn't be used anymore directly as `dma_wait` should be more accessible and then `column_num` and `row_num` would only be set in a couple of places. We could also make these arguments optional and default to 1.

Reply:
There's a similar `column_num` field in the `writebd_shimtile` instruction. None of these map to bits in hardware registers. Instead they are a firmware feature to write to multiple tiles with one instruction. I don't know of anyone using the them with mlir-aie.